sui_core/src/authority.rs

```diff
@@ -806,6 +815,145 @@ impl AuthorityState {
+                    match self.database.get_object_version(object_id, obj_ref.1)? {
+                        None => {
+                            error!("Object with in parent_entry is missing from object store, datastore is inconsistent");
+                            Err(SuiError::ObjectNotFound {
+                                object_id: *object_id,
+                            })
+                        }
+                        Some(object) => {
+                            let resolver = ModuleCache::new(&self);
+                            let layout =
+                                object.get_layout(ObjectFormatOptions::default(), &resolver)?;
+                            Ok(ObjectRead::Exists(obj_ref, object, layout))
+                        }
+                    }
+                } else {
+                    Ok(ObjectRead::Deleted(obj_ref))
+                }
+            }
+        }
+    }
+
+    pub async fn get_owned_objects(&self, account_addr: SuiAddress) -> SuiResult<Vec<ObjectRef>> {
+        self.database.get_account_objects(account_addr)
+    }
+
+    pub fn get_total_transaction_number(&self) -> Result<u64, anyhow::Error> {
+        Ok(self.database.next_sequence_number()?)
+    }
+
+    pub fn get_transactions_in_range(
+        &self,
+        start: TxSequenceNumber,
+        end: TxSequenceNumber,
+    ) -> Result<Vec<(TxSequenceNumber, TransactionDigest)>, anyhow::Error> {
+        fp_ensure!(
+            start <= end,
+            SuiError::GatewayInvalidTxRangeQuery {
+                error: format!(
+                    "start must not exceed end, (start={}, end={}) given",
+                    start, end
+                ),
+            }
+            .into()
+        );
+        fp_ensure!(
+            end - start <= MAX_TX_RANGE_SIZE,
+            SuiError::GatewayInvalidTxRangeQuery {
+                error: format!(
+                    "Number of transactions queried must not exceed {}, {} queried",
+                    MAX_TX_RANGE_SIZE,
+                    end - start
+                ),
+            }
+            .into()
+        );
+        let res = self.database.transactions_in_seq_range(start, end)?;
+        debug!(?start, ?end, ?res, "Fetched transactions");
+        Ok(res)
+    }
+
+    pub fn get_recent_transactions(
+        &self,
+        count: u64,
+    ) -> Result<Vec<(TxSequenceNumber, TransactionDigest)>, anyhow::Error> {
+        fp_ensure!(
+            count <= MAX_TX_RANGE_SIZE,
+            SuiError::GatewayInvalidTxRangeQuery {
+                error: format!(
+                    "Number of transactions queried must not exceed {}, {} queried",
+                    MAX_TX_RANGE_SIZE, count
+                ),
+            }
+            .into()
+        );
+        let end = self.get_total_transaction_number()?;
+        let start = if end >= count { end - count } else { 0 };
+        self.get_transactions_in_range(start, end)
+    }
+
+    pub async fn get_transaction(
```

Comment:
Actually, this one probably can be replaced with make_transaction_info.

Reply:
It can, but it doesn't really reduce the amount of code because of all the Option handling you have to do.